common/persistence/historyStore.go

```diff
@@ -169,8 +171,12 @@ func (m *historyV2ManagerImpl) GetHistoryTree(
 	if err != nil {
 		return nil, err
 	}
+	var branches []*workflow.HistoryBranch
```

Comment:
nit: initialize with known capacity to avoid resizing - `make([]*workflow.HistoryBranch, 0, len(resp.Branches))`

Reply:
This will not handle nil correctly without adding another check.